docs/dev/framework/caching.md

```diff
@@ -217,24 +217,104 @@ Moreover, you don't have to register to all the different callbacks such as `ons
+only for one hour, the whole page will only be cached for one hour.
+
+A more common use case would be something like a `{{date::Y}}` insert tag, which renders the current year. This
+fragment is cacheable until the end of 31st of December. If the page is cached on the 10th of December for one day,
+the page cache time will not be affected. However, if the page is cached at 12pm on the 31st of December, the cache
+time will be lowered to 12 instead of 24 hours. The page cache will expire at the end of the year and regenerated
+on the 1st of January.
+
+
+{{%expand "How this works under the hood" %}}
+A `Response` object in Symfony is `Cache-Control: private` by default. This means we cannot *always* merge an
+inline fragment, otherwise all pages would be uncacheable by default. To work around this, we've added a little
+trick: the response must have a specific header to be merged.
+
+```php
+use Symfony\Component\HttpFoundation\Response;
+
+/** @var Response $response */
+$response->headers->set('Contao-Merge-Cache-Control', true);
+```
+
+This *trick* is automatically applied if a response is generated from a Contao template. A fragment controller
+can affect the cache lifetime by setting it on the template response:
+
+```php
+// src/Controller/MySuperController.php
+namespace App\Controller;
+
+use Contao\CoreBundle\Controller\FrontendModule\AbstractFrontendModuleController;
+use Contao\ModuleModel;
+use Contao\Template;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+class MySuperController extends AbstractFrontendModuleController
+{
+    protected function getResponse(Template $template, ModuleModel $model, Request $request): ?Response
+    {
+        $response = $template->getResponse();
+        
+        $response->setPublic();
+        $response->setMaxAge(3600);
+        
+        return $response;
+    }
+}
+```
+{{% /expand%}}
+
+
+### Edge Side Includes (ESI)
+
+By setting the fragment renderer to `esi`, the fragment will be cached **separately** from the main content.
+The page can be cached for 24 hours, but the frament can be cached for one week. On every request to the page,
+the reverse proxy will merge the two pieces, rebuilding each if its cache has expired.
+
+A common use case for this is a fragment that can certainly be cached longer than the current page, but is
+expensive to generate. Something like a weather preview, which requires an API request, but only updates once a day.
+Do not use ESI if your fragment is inexpensive to generate (like the `{{date::Y}}` insert tag). For inexpensive
+cases, it is most likely better to cache the whole page and **re-generate more often** than having to merge multiple
+fragments **on each request**.
+
+Symfony automatically detects if it is talking to  a gateway cache that supports ESI
+(like Symfony's built in reverse proxy, that Contao uses). ESI is also supported by
```

Comment:
Only the Managed Edition.

Reply:
yeah… thats how the docs were before. Not sure if that detail matters?

Reply:
It does.

Reply:
Added in 204e48085716e72b5a31785de634f913592f381b